rpc/server.go

```diff
@@ -81,8 +81,8 @@ func (s *Server) ServeCodec(codec ServerCodec, options CodecOption) {
 	}
 
 	// Add the codec to the set so it can be closed by Stop.
-	s.codecs.Add(codec)
-	defer s.codecs.Remove(codec)
+	s.codecs.Add(&codec)
```

Comment:
This change looks like it changes the semantics, since it now adds the pointer to a codec instead of the codec struct. @fjl PTAL to double-check if this is ok

Reply:
I will fix this in a follow-up PR. Using a set here is weird.

Reply:
Yes. This is one the cases where Go generics codifies something that couldn't be expressed in types before. Go allows interface values to be used as map keys, but will panic at runtime if the value contained in the interface is not comparable. With the generic set, this is not possible anymore. There is also no easy way to turn ServerCodec into something 'comparable'.

Have submitted https://github.com/ethereum/go-ethereum/pull/26180 to remove the pointer.